internal/tracinginterceptor/interceptor_test.go

```diff
@@ -19,3 +19,151 @@
 // THE SOFTWARE.
 
 package tracinginterceptor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/opentracing/opentracing-go/mocktracer"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/yarpc/api/transport"
+)
+
+// Define UnaryHandlerFunc to adapt a function into a UnaryHandler.
+type UnaryHandlerFunc func(ctx context.Context, req *transport.Request, resw transport.ResponseWriter) error
+
+func (f UnaryHandlerFunc) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter) error {
+	return f(ctx, req, resw)
+}
+
+// Define OnewayHandlerFunc to adapt a function into a OnewayHandler.
+type OnewayHandlerFunc func(ctx context.Context, req *transport.Request) error
+
+func (f OnewayHandlerFunc) HandleOneway(ctx context.Context, req *transport.Request) error {
+	return f(ctx, req)
+}
+
+// Define UnaryOutboundFunc to adapt a function into a UnaryOutbound.
+type UnaryOutboundFunc func(ctx context.Context, req *transport.Request) (*transport.Response, error)
+
+func (f UnaryOutboundFunc) Call(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+	return f(ctx, req)
+}
+
+// Implement Start for UnaryOutboundFunc (No-op for testing purposes)
+func (f UnaryOutboundFunc) Start() error {
+	return nil
+}
+
+// Implement Stop for UnaryOutboundFunc (No-op for testing purposes)
+func (f UnaryOutboundFunc) Stop() error {
+	return nil
+}
+
+// Implement IsRunning for UnaryOutboundFunc (Returns false for testing purposes)
+func (f UnaryOutboundFunc) IsRunning() bool {
+	return false
+}
+
+// Implement Transports for UnaryOutboundFunc (Returns nil for testing purposes)
+func (f UnaryOutboundFunc) Transports() []transport.Transport {
+	return nil
+}
+
+// Setup mock tracer
+func setupMockTracer() *mocktracer.MockTracer {
+	return mocktracer.New()
+}
+
+// TestUnaryInboundHandle tests the Handle method for Unary Inbound
+func TestUnaryInboundHandle(t *testing.T) {
```

Comment:
Can we use the [test table](https://github.com/uber-go/guide/blob/master/style.md#test-tables) patter to test it more comprehensively like testing against successful case, failed case 1 (error is not nil), failed case 2(error is nil but with application error)? 

Same for outbound.